cirq/contrib/quantum_volume/quantum_volume.py

```diff
@@ -422,12 +423,22 @@ def execute_circuits(
     return results
 
 
+@deprecated_parameter(
+    deadline="v0.12",
+    fix="use device_qubits instead",
+    parameter_desc='device_or_qubits',
+    match=lambda args, kwargs: 'device_or_qubits' in kwargs,
+    rewrite=lambda args, kwargs: (
+        args,
+        {('device_qubits' if k == 'device_or_qubits' else k): v for k, v in kwargs.items()},
+    ),
+)
 def calculate_quantum_volume(
     *,
     num_qubits: int,
     depth: int,
     num_circuits: int,
-    device_or_qubits: Union[cirq.google.XmonDevice, List[cirq.GridQubit]],
+    device_qubits: List[cirq.GridQubit],
```

Comment:
can we change this to `device_graph: nx.Graph`? All of Cirq is going to come crashing down the instant we have a device with a broken coupler. An immediate benefit is supporting qubits other than gridqubit.

Reply:
I'm down - I actually really like the lightweight implementation in the routing package for the graph device...

Reply:
I use it all the time (once you internalize the sometimes idiosyncratic naming of the functions). See here for a related issue: https://github.com/quantumlib/Cirq/issues/3833

gridqubits_to_graph_device is implemented fairly inefficiently at present.